spec/std/json/serializable_spec.cr

```diff
@@ -465,6 +465,18 @@ module JSONNamespace
   end
 end
 
+class JSONSomething
+  include JSON::Serializable
+
+  property value : JSONAttrValue(Set(JSONSomethingElse)?)?
```

Comment:
`JSONAttrValue()` wrapper is not required here for the issue to manifest. Why it was added? Doesn't that add possibility to hide the issue in the future (for some other bugs)?

Reply:
Good catch. The spec setup can actually be simplified a bit:
```diff
--- i/spec/std/json/serializable_spec.cr
+++ w/spec/std/json/serializable_spec.cr
@@ -468,13 +468,7 @@ end
 class JSONSomething
   include JSON::Serializable

-  property value : JSONAttrValue(Set(JSONSomethingElse)?)?
-end
-
-class JSONSomethingElse
-  include JSON::Serializable
-
-  property value : JSONAttrValue(Set(JSONSomethingElse)?)?
+  property value : JSONSomething?
 end

 describe "JSON mapping" do
@@ -1122,6 +1116,6 @@ describe "JSON mapping" do

   it "fixes #13337" do
     JSONSomething.from_json(%({"value":{}})).value.should_not be_nil
-    JSONSomethingElse.from_json(%({"value":{}})).value.should_not be_nil
+    JSONAttrValue(JSONSomething).from_json(%({"value":{}})).value.should_not be_nil
   end
 end
 ```

Reply:
IIRC this wouldn't reproduce the bug without this patch

Reply:
No I checked that it actually does reproduce.

Reply:
I'm still going to merge this as it to move along with 1.8.1.
We can refactor the specs afterwards.

Reply:
https://github.com/crystal-lang/crystal/pull/13358